include/alpaka/math/max/MaxStdLib.hpp

```diff
@@ -27,32 +27,21 @@ namespace alpaka
 
         namespace traits
         {
-            //! The standard library integral max trait specialization.
+            //! The standard library max trait specialization.
             template<typename Tx, typename Ty>
-            struct Max<MaxStdLib, Tx, Ty, std::enable_if_t<std::is_integral<Tx>::value && std::is_integral<Ty>::value>>
-            {
-                ALPAKA_FN_HOST auto operator()(MaxStdLib const& max_ctx, Tx const& x, Ty const& y)
-                {
-                    alpaka::ignore_unused(max_ctx);
-                    using std::max;
-                    return max(x, y);
-                }
-            };
-            //! The standard library mixed integral floating point max trait specialization.
-            template<typename Tx, typename Ty>
-            struct Max<
-                MaxStdLib,
-                Tx,
-                Ty,
-                std::enable_if_t<
-                    std::is_arithmetic<Tx>::value && std::is_arithmetic<Ty>::value
-                    && !(std::is_integral<Tx>::value && std::is_integral<Ty>::value)>>
+            struct Max<MaxStdLib, Tx, Ty, std::enable_if_t<std::is_arithmetic_v<Tx> && std::is_arithmetic_v<Ty>>>
             {
                 ALPAKA_FN_HOST auto operator()(MaxStdLib const& max_ctx, Tx const& x, Ty const& y)
                 {
                     alpaka::ignore_unused(max_ctx);
+
                     using std::fmax;
-                    return fmax(x, y);
+                    using std::max;
+
+                    if constexpr(std::is_integral_v<Tx> && std::is_integral_v<Ty>)
+                        return max(x, y);
+                    else // Floating point or mixed integral-floating point
```

Comment:
@psychocoderHPC How do we want to handle the standard library math functions? Should we impose the same restrictions as for CUDA devices (no `long double`)?

Reply:
IMO we should forbid `long double` until it became important for one of our users. IMO the current scope of alpaka is portability and this is easier if `long double` can not be used.

Reply:
INdependent of what you choose but please remove the `else` as in all other cases, that we are always explicit.